error: could not compile `neqo-common` due to 4 previous errors

I am following the linux build and dockerfile for building but keep stuck at this error when doing cargo build.

error: could not compile `neqo-common` due to 4 previous errors

```
error[E0061]: this function takes 4 arguments but 3 arguments were supplied
    --> neqo-common/src/log.rs:16:13
     |
16   |             ::log::__private_api_log(
     |             ^^^^^^^^^^^^^^^^^^^^^^^^ expected 4 arguments
17   |                 ::log::__log_format_args!($($arg)+),
18   |                 lvl,
     |                 ---
19   |                 &($target, ::log::__log_module_path!(), ::log::__log_file!(), ::log::__log_line!()),
     |                 ----------------------------------------------------------------------------------- supplied 3 arguments
...
60   |             do_log!(::log::Level::Info, "Logging initialization error {:?}", e);
     |             -------------------------------------------------------------------
     |             |
     |             in this macro invocation
     |
note: function defined here
    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.16/src/lib.rs:1576:8
     |
1576 | pub fn __private_api_log(
     |        ^^^^^^^^^^^^^^^^^
     = note: this error originates in the macro `$crate::do_log` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0061`.
error: could not compile `neqo-common` due to 4 previous errors

Caused by:
  process didn't exit successfully: `rustc --crate-name neqo_common --edition=2018 neqo-common/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="deny-warnings"' -C metadata=cbd48452ed35b24a -C extra-filename=-cbd48452ed35b24a --out-dir /src/neqo/target/debug/deps -C incremental=/src/neqo/target/debug/incremental -L dependency=/src/neqo/target/debug/deps --extern chrono=/src/neqo/target/debug/deps/libchrono-4896f07c16cee554.rmeta --extern env_logger=/src/neqo/target/debug/deps/libenv_logger-c1ca2f2bb9193a60.rmeta --extern lazy_static=/src/neqo/target/debug/deps/liblazy_static-02ff0d0fbc57b9b4.rmeta --extern log=/src/neqo/target/debug/deps/liblog-38ede5edb5cb0d57.rmeta --extern qlog=/src/neqo/target/debug/deps/libqlog-99ded8192d30c386.rmeta` (exit status: 1)
warning: build failed, waiting for other jobs to finish...
error: build failed
```

do I miss something?
please help and thank you.

--- a/neqo-common/Cargo.toml
+++ b/neqo-common/Cargo.toml
@@ -8,7 +8,7 @@ license = "MIT/Apache-2.0"
 build = "build.rs"
 
 [dependencies]
-log = {version = "0.4.0", default-features = false}
+log = {version = "0.4.16", default-features = false}
 env_logger = {version = "0.8", default-features = false}
 lazy_static = "1.3.0"
 qlog = "0.4.0"

--- a/neqo-common/src/log.rs
+++ b/neqo-common/src/log.rs
@@ -17,6 +17,7 @@ macro_rules! do_log {
                 ::log::__log_format_args!($($arg)+),
                 lvl,
                 &($target, ::log::__log_module_path!(), ::log::__log_file!(), ::log::__log_line!()),
+                None,
             );
         }
     });

